addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiDeviceBaseHandler.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mihome.handler;
+
+import static org.openhab.binding.mihome.XiaomiGatewayBindingConstants.*;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.mihome.internal.XiaomiItemUpdateListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link XiaomiDeviceBaseHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Patrick Boos - Initial contribution
+ * @author Kuba Wolanin - Added voltage and low battery report
+ * @author Dimalo - Added cube rotation, heartbeat and voltage handling, configurable window and motion delay, Aqara
+ *         switches
+ */
+public abstract class XiaomiDeviceBaseHandler extends BaseThingHandler implements XiaomiItemUpdateListener {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(
+            Arrays.asList(THING_TYPE_GATEWAY, THING_TYPE_SENSOR_HT, THING_TYPE_SENSOR_MOTION, THING_TYPE_SENSOR_SWITCH,
+                    THING_TYPE_SENSOR_MAGNET, THING_TYPE_SENSOR_CUBE, THING_TYPE_SENSOR_AQARA1,
+                    THING_TYPE_SENSOR_AQARA2, THING_TYPE_ACTOR_AQARA1, THING_TYPE_ACTOR_AQARA2, THING_TYPE_ACTOR_PLUG));
+
+    private static final long ONLINE_TIMEOUT = 2 * 60 * 60 * 1000; // 2 hours
+
+    private JsonParser parser = new JsonParser();
+
+    private XiaomiBridgeHandler bridgeHandler;
+
+    String itemId;
+
+    Logger logger = LoggerFactory.getLogger(this.getClass().getName());
+
+    public XiaomiDeviceBaseHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        final String configItemId = (String) getConfig().get(ITEM_ID);
+        if (configItemId != null) {
```

Comment:
Why do you need this check what value do you think that `itemId` will have now?

imho this should just be

```java
itemId = (String) getConfig().get(ITEM_ID);
```